PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/runner/functions/localrun/FunctionRunState.java

```diff
@@ -107,17 +127,72 @@ private void launchDebugger(final Project project, int debugPort) {
     protected FunctionApp executeSteps(@NotNull RunProcessHandler processHandler, @NotNull Map<String, String> telemetryMap) throws Exception {
         // Prepare staging Folder
         updateTelemetryMap(telemetryMap);
+        validateFunctionRuntime(processHandler);
         stagingFolder = FunctionUtils.getTempStagingFolder();
         prepareStagingFolder(stagingFolder, processHandler);
         // Run Function Host
         runFunctionCli(processHandler, stagingFolder);
         return null;
     }
 
-    private void runFunctionCli(RunProcessHandler processHandler, File stagingFolder) throws IOException, InterruptedException {
+    private void validateFunctionRuntime(RunProcessHandler processHandler) throws AzureExecutionException {
+        try {
+            final String funcPath = functionRunConfiguration.getFuncPath();
+            if (StringUtils.isEmpty(funcPath)) {
+                throw new AzureExecutionException(RUNTIME_NOT_FOUND);
+            }
+            final ComparableVersion funcVersion = getFuncVersion();
+            if (funcVersion == null) {
+                throw new AzureExecutionException(RUNTIME_NOT_FOUND);
+            }
+            final ComparableVersion javaVersion = getJavaVersion();
+            if (javaVersion == null) {
+                processHandler.setText(FAILED_TO_GET_JAVA_VERSION);
+                return;
+            }
+            if (javaVersion.compareTo(JAVA_9) < 0) {
+                // No need validate function host version within java 8 or earlier
+                return;
+            }
+            final ComparableVersion minimumVersion = funcVersion.compareTo(FUNC_3) >= 0
+                                                     ? MINIMUM_JAVA_9_SUPPORTED_VERSION
+                                                     : MINIMUM_JAVA_9_SUPPORTED_VERSION_V2;
+            if (funcVersion.compareTo(minimumVersion) < 0) {
+                throw new AzureExecutionException(FUNCTION_CORE_TOOLS_OUT_OF_DATE);
+            }
+        } catch (IOException | InterruptedException e) {
+            processHandler.setText(String.format(FAILED_TO_VALIDATE_FUNCTION_RUNTIME, e.getMessage()));
+        }
+    }
+
+    private ComparableVersion getFuncVersion() throws IOException, InterruptedException {
+        final String[] funcVersionResult = CommandUtils.executeMultipleLineOutput(
+                String.format("%s -v", functionRunConfiguration.getFuncPath()), null);
```

Comment:
String.format("%s -v... will cause the func -v not work when the path contains spaces.